backend/service/workflow/github.go

```diff
@@ -0,0 +1,65 @@
+package workflow
+
+import (
+	"net/http"
+	"net/url"
+
+	v1 "github.com/ekspand/trusty/api/v1"
+	"github.com/ekspand/trusty/internal/db/model"
+	"github.com/go-phorce/dolly/xhttp/marshal"
+	"github.com/google/go-github/github"
+	"github.com/juju/errors"
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2"
+)
+
+// SyncGithubRepos retrieves Repos and Orgs info for the user from github
+func (s *Service) SyncGithubRepos(ctx context.Context, w http.ResponseWriter, user *model.User) error {
+	o := s.OAuthConfig(user.Provider)
+
+	conf := &oauth2.Config{
+		ClientID:     o.ClientID,
+		ClientSecret: o.ClientSecret,
+		RedirectURL:  s.cfg.TrustyClient.PublicURL + v1.PathForAuthGithubCallback,
+		Scopes:       o.Scopes,
+		Endpoint: oauth2.Endpoint{
+			AuthURL:  o.AuthURL,
+			TokenURL: o.TokenURL,
+		},
+	}
+
+	token := &oauth2.Token{
+		AccessToken:  user.AccessToken,
+		RefreshToken: user.RefreshToken,
+		TokenType:    "Bearer",
+		Expiry:       user.TokenExpiresAt.Time,
+	}
+
+	client := github.NewClient(conf.Client(ctx, token))
+	if s.GithubBaseURL != "" {
+		if u, err := url.Parse(s.GithubBaseURL); err == nil {
```

Comment:
if err is nil, should not we log and stop the method execution?

Reply:
No. This `GithubBaseURL` is used in tests only when we don't need to hit the Prod URL

Reply:
I think even if is used only for tests it is right to log the error and stop the method. What you are saying is that anyways this code path won't be hit other than in tests. However for the readers of this code clearly that is not apparent.

Reply:
agree. let's do that